agent_test.go

```diff
@@ -153,3 +154,161 @@ func TestAgentConfigGetConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestReadCtl(t *testing.T) {
+	f, err := ioutil.TempFile("", "ctl")
+	if err != nil {
+		t.Error(err)
+	}
+	defer f.Close()
+	defer os.Remove(f.Name())
+
+	hCmd := uint32(5)
+	p := &pod{ctl: f}
+
+	// file is empty, read should fail
+	_, _, err = p.readCtl()
+	if err == nil {
+		t.Error(err)
+	}
+
+	// no enough bytes, read should fail
+	buff := make([]byte, ctlHeaderSize-1)
+	f.Write(buff)
+	f.Seek(0, 0)
+	_, _, err = p.readCtl()
+	if err == nil {
+		t.Error(err)
+	}
+
+	// there is no data, read should fail
+	buff = make([]byte, ctlHeaderSize)
+	f.Seek(0, 0)
+	binary.BigEndian.PutUint32(buff[4:ctlHeaderSize], ctlHeaderSize+1)
+	f.Write(buff)
+	f.Seek(0, 0)
+	_, _, err = p.readCtl()
+	if err == nil {
+		t.Error(err)
+	}
+
+	bufferSizes := []uint32{0, 2, 55, 278, 698, 1056, 3598, 6527, 10830, 56386, 894305, 9652381}
+	for _, s := range bufferSizes {
+		f.Seek(0, 0)
+
+		// write command and length
+		buff = make([]byte, ctlHeaderSize)
+		binary.BigEndian.PutUint32(buff[:4], hCmd)
+		binary.BigEndian.PutUint32(buff[4:ctlHeaderSize], s+ctlHeaderSize)
+		n, err := f.Write(buff)
+		if err != nil {
+			t.Error(err)
+		}
+		if uint32(n) != uint32(ctlHeaderSize) {
+			t.Error("expected: %d, actual: %d", n, ctlHeaderSize)
```

Comment:
Travis is failing as the linters have detected this and a few other calls should be `t.Errorf()`.

Reply:
oops!, fixed, thanks